design/benchmark_tests.md

```diff
@@ -0,0 +1,30 @@
+# Benchmarks Tests
+
+## Requirements
+1. Measure the latency percentiles (50th, 90th, 95th, 99th 99.9) for a batch of repeated basic operations (e.g. 10K put requests)  
+2. Show trends of benchmarks tests execution times across different versions.
+
+## Non-Requirements
+1. An active gatekeeper - single slow running cycle shouldn't block the build.
+2. Expect 100% reliability of all actions - when performing thousands of operations,
```

Comment:
Think that success is part of what we measure - it can be indicator for the system test. But usually when you test benchmark you can hit a limit and sometimes the limit is when things starts to fail.

Reply:
We can leverage the prometheus metrics from the lakeFS app that will reflect the status codes of the requests. Is that enough or did you mean to collect more metrics from the test itself? Because that means that it would have to collect and publish metrics too.

Reply:
Just the part of the 'requirement' that we need to verify that we report X performance when the system does the actual work. It can work faster sometimes when all requests fails.

Reply:
Added.